### ISSUE TITLE
Pulsar IO tests leak resources in Kafka related tests

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#15853]
**Describe the bug**

Resources are leaked in Pulsar IO tests.

This type of entries repeat in the test logs:
```
2022-05-31T06:06:42,491 - WARN  - [kafka-coordinator-heartbeat-thread | sink-test-fyeeijxp:NetworkClient@782] - [Consumer clientId=consumer-sink-test-fyeeijxp-1, groupId=sink-test-fyeeijxp] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:42,674 - WARN  - [kafka-producer-network-thread | c9eafaa1-fee0-4029-a05d-55b0c639cb3e:NetworkClient@782] - [Producer clientId=c9eafaa1-fee0-4029-a05d-55b0c639cb3e] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:43,367 - WARN  - [kafka-coordinator-heartbeat-thread | sink-test-fyeeijxp:NetworkClient@782] - [Consumer clientId=consumer-sink-test-fyeeijxp-1, groupId=sink-test-fyeeijxp] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:43,834 - WARN  - [kafka-producer-network-thread | c9eafaa1-fee0-4029-a05d-55b0c639cb3e:NetworkClient@782] - [Producer clientId=c9eafaa1-fee0-4029-a05d-55b0c639cb3e] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:44,359 - WARN  - [kafka-coordinator-heartbeat-thread | sink-test-fyeeijxp:NetworkClient@782] - [Consumer clientId=consumer-sink-test-fyeeijxp-1, groupId=sink-test-fyeeijxp] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:44,748 - WARN  - [kafka-producer-network-thread | c9eafaa1-fee0-4029-a05d-55b0c639cb3e:NetworkClient@782] - [Producer clientId=c9eafaa1-fee0-4029-a05d-55b0c639cb3e] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:45,285 - WARN  - [kafka-coordinator-heartbeat-thread | sink-test-fyeeijxp:NetworkClient@782] - [Consumer clientId=consumer-sink-test-fyeeijxp-1, groupId=sink-test-fyeeijxp] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:45,603 - WARN  - [kafka-producer-network-thread | c9eafaa1-fee0-4029-a05d-55b0c639cb3e:NetworkClient@782] - [Producer clientId=c9eafaa1-fee0-4029-a05d-55b0c639cb3e] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:46,457 - WARN  - [kafka-coordinator-heartbeat-thread | sink-test-fyeeijxp:NetworkClient@782] - [Consumer clientId=consumer-sink-test-fyeeijxp-1, groupId=sink-test-fyeeijxp] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:46,708 - WARN  - [kafka-producer-network-thread | c9eafaa1-fee0-4029-a05d-55b0c639cb3e:NetworkClient@782] - [Producer clientId=c9eafaa1-fee0-4029-a05d-55b0c639cb3e] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:47,494 - WARN  - [kafka-coordinator-heartbeat-thread | sink-test-fyeeijxp:NetworkClient@782] - [Consumer clientId=consumer-sink-test-fyeeijxp-1, groupId=sink-test-fyeeijxp] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:47,863 - WARN  - [kafka-producer-network-thread | c9eafaa1-fee0-4029-a05d-55b0c639cb3e:NetworkClient@782] - [Producer clientId=c9eafaa1-fee0-4029-a05d-55b0c639cb3e] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
2022-05-31T06:06:48,409 - WARN  - [kafka-coordinator-heartbeat-thread | sink-test-fyeeijxp:NetworkClient@782] - [Consumer clientId=consumer-sink-test-fyeeijxp-1, groupId=sink-test-fyeeijxp] Connection to node 1 (localhost/127.0.0.1:49216) could not be established. Broker may not be available.
```

**To Reproduce**

Check the output of "[CI - System - Pulsar IO](https://github.com/apache/pulsar/runs/6664811881?check_suite_focus=true#logs)" 

**Expected behavior**

Resources should be properly closed in tests using try-with-resources blocks or Lombok's `@Cleanup` annotation.

**Comment by lhotari** [comment]
I'll create a PR with these changes: https://github.com/apache/pulsar/compare/master...lhotari:lh-fix-pulsar-io-integration-test-resource-leaks after CI is unblocked.

**Comment by lhotari** [comment]
This problem isn't fixed. example [CI - System - Pulsar IO](https://github.com/apache/pulsar/actions/runs/3541194507/jobs/5947930088) job 
```
2022-11-24T16:43:26.5108101Z 2022-11-24T16:43:26,510 - WARN  - [kafka-producer-network-thread | 2f374e07-6664-47d6-91c6-26f846bf509b:NetworkClient@782] - [Producer clientId=2f374e07-6664-47d6-91c6-26f846bf509b] Connection to node 1 (localhost/127.0.0.1:49232) could not be established. Broker may not be available.
2022-11-24T16:43:27.5136085Z 2022-11-24T16:43:27,513 - WARN  - [kafka-producer-network-thread | 2f374e07-6664-47d6-91c6-26f846bf509b:NetworkClient@782] - [Producer clientId=2f374e07-6664-47d6-91c6-26f846bf509b] Connection to node 1 (localhost/127.0.0.1:49232) could not be established. Broker may not be available.
2022-11-24T16:43:28.4162827Z 2022-11-24T16:43:28,415 - WARN  - [kafka-producer-network-thread | 2f374e07-6664-47d6-91c6-26f846bf509b:NetworkClient@782] - [Producer clientId=2f374e07-6664-47d6-91c6-26f846bf509b] Connection to node 1 (localhost/127.0.0.1:49232) could not be established. Broker may not be available.
2022-11-24T16:43:29.6195148Z 2022-11-24T16:43:29,619 - WARN  - [kafka-producer-network-thread | 2f374e07-6664-47d6-91c6-26f846bf509b:NetworkClient@782] - [Producer clientId=2f374e07-6664-47d6-91c6-26f846bf509b] Connection to node 1 (localhost/127.0.0.1:49232) could not be established. Broker may not be available.
2022-11-24T16:43:30.8218768Z 2022-11-24T16:43:30,821 - WARN  - [kafka-producer-network-thread | 2f374e07-6664-47d6-91c6-26f846bf509b:NetworkClient@782] - [Producer clientId=2f374e07-6664-47d6-91c6-26f846bf509b] Connection to node 1 (localhost/127.0.0.1:49232) could not be established. Broker may not be available.
```

**Comment by lhotari** [comment]
There was an additional thread leak which will be fixed by #18618
